src/components/Post/PostDetailComponent.jsx

```diff
@@ -1,16 +1,128 @@
 import React, { Component } from 'react';
-import { Segment } from 'semantic-ui-react';
+import { Segment, Image } from 'semantic-ui-react';
+import { withRouter, Link } from 'react-router-dom';
+import GoogleMapReact from 'google-map-react';
+
+import MapMarker from '../Home/MapMarkers';
 
 class NewPost extends Component {
+  state = {
+    zoom:15,
+  }
+  
+  postColorHandler = (post) => {
+    let postColor = '';
+    if (post.user_rating === 'Gnarly') {
+      postColor = 'red';
+    } else if (post.user_rating === 'Good') {
+      postColor = 'orange';
+    } else if (post.user_rating === 'Fair') {
+      postColor = 'yellow';
+    } else if (post.user_rating === 'Poor') {
+      postColor = 'teal';
+    } else if (post.user_rating === 'Flat') {
+      postColor = 'grey';
+    }
+    return postColor;
+  };
+  
+  getPost = async () => {
+    if (this.state.post === undefined) {
+      const response = await fetch(`${process.env.REACT_APP_DEV_API_DOMAIN}/posts/${this.props.match.params.post_id.toString()}`);
+      const { post } = await response.json();
+      this.setState({ post: post, userLocation: { lat: post.lat, lng: post.lng } });
+      localStorage.setItem('post_user_Id', this.state.post.user.id);
+    } 
+  }
+  
   render() {
+    this.getPost();
     return (
       <Segment className="new-post-ocean-pic">
-        <Segment className="home-row" stacked>
-          <div>P o s t_D e t a i l_</div>
-        </Segment>
+        {
+          this.state.post !== undefined
+            ?(
+              <Segment className="home-row" stacked>
+                <Segment className='post-detail-div'>
+                  <h1>
+                    P O S T_D E T A I L_
+                  </h1>
+                  <p>
+                    U S E R_S T O K E_R A T I N G: {this.state.post.user_rating}
+                  </p>
+                  <Image src='https://i.pinimg.com/originals/90/64/a1/9064a16dc937d44f1fedba59074d5fa5.jpg' size='large' wrapped />
+                  <h4>
+                    p o s t e d_b y: {this.state.post.user.username}
+                  </h4>
+                  <div className='break-word'>
+                    {this.state.post.post_content}
+                  </div>
+                </Segment>
+                <Segment id='post-detail-map-div'>
+                  <h3>
+                    {this.state.post.city},
+                    {' ' + this.state.post.state}
```

Comment:
Rather than doing this, you can just do `{\`${this.state.post.city}, ${this.state.post.state}\`}`